crates/fuel-core/src/query/message.rs

```diff
@@ -205,7 +205,12 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
         None => return Ok(None),
     };
 
-    let verifiable_commit_block_height = *commit_block_header.height() - 1u32.into();
+    let block_heigth = *commit_block_header.height();
+    if block_heigth == 0u32.into() {
```

Comment:
Hmm, actually, you should be able to use Genesis Block to generate proofs, but because it is a specific use case, we need to define how we want to do that and what the proof should look like. 

It is not related to this PR and should be solved during https://github.com/FuelLabs/fuel-core/issues/816